### DISCUSSION TITLE
What are the valid env.targets?

### DISCUSSION THREAD

**Discussion by patricklafrance** [swc-project/swc#7855] [Q&A] [answered]
I am implementing a function to convert browserslist results to SWC targets, similar to https://github.com/marcofugaro/browserslist-to-esbuild/tree/main. I keep getting error messages like the following:

```
thread '<unnamed>' panicked at 'not implemented: Targets: {"bb": Query(Single("7")), "ie": Query(Single("5.5")), "edge": Query(Single("106")), "firefox": Query(Single("107")), "opera": Query(Single("92")), "android": Query(Single("115")), "chrome": Query(Single("107")), "samsung": Query(Single("12")), "safari": Query(Single("15.2"))}', C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\preset_env_base-0.4.5\src\query.rs:124:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
assets by status 1.1 KiB [cached] 2 assets
./src/index.ts 39 bytes [built] [code generated] [1 error]
```

The message seem to happens because I am providing an invalid target. For example "baidu", "kaios" or "bb" are not supported.

According to SWC documentation, the following environments are supported:

- chrome
- opera
- edge
- firefox
- safari
- ie
- ios
- android
- node
- electron

But it is stated that they are "example" environments.

I am wondering where I could find a list of ALL the supported environments?

Thank you!

Patrick

**Comment by kwonoj** [comment]
We use browserslist-rs to query those, so best bet is looking into browserslist-rs supported lists.

**Comment by patricklafrance** [comment]
@kwonoj Thank for your answer. Sadly, browserslist-rs doesn't provide much documentation and nothing is mentionned about the supported list.

Definitely, it doesn't support the same list as the original browserslist tool.

**Comment by patricklafrance** [comment]
I opened an issue in the browserslist-rs repository, maybe I'll get an answer: https://github.com/browserslist/browserslist-rs/issues/20

**Comment by patricklafrance** [comment, marked as answer]
If someone is looking for the list, it's here: [browserslist-rs/build.rs](https://github.com/browserslist/browserslist-rs/blob/99a3244fc8c0e631a80a9cae5c41dca6c5a2aae5/build.rs#L10)
